taraxa/state/dpos/precompiled/dpos_contract.go

```diff
@@ -0,0 +1,1022 @@
+		return nil, self.redelegate(ctx, evm.GetBlock().Number, args)
+
+	case "claimRewards":
+		var args ValidatorAddress
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse claimRewards input args: ", err)
+			return nil, err
+		}
+
+		return nil, self.claimRewards(ctx, evm.GetBlock().Number, args)
+
+	case "claimCommissionRewards":
+		return nil, self.claimCommissionRewards(ctx, evm.GetBlock().Number)
+
+	case "registerValidator":
+		var args RegisterValidatorArgs
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse registerValidator input args: ", err)
+			return nil, err
+		}
+
+		return nil, self.registerValidator(ctx, evm.GetBlock().Number, args)
+
+	case "setValidatorInfo":
+		var args SetValidatorInfoArgs
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse setValidatorInfo input args: ", err)
+			return nil, err
+		}
+
+		return nil, self.setValidatorInfo(ctx, args)
+
+	case "isValidatorEligible":
+		var args ValidatorAddress
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse isValidatorEligible input args: ", err)
+			return nil, err
+		}
+		result := self.delayedStorage.IsEligible(&args.Validator)
+		return method.Outputs.Pack(result)
+
+	case "getTotalEligibleValidatorsCount":
+		result := self.delayedStorage.EligibleAddressCount()
+		return method.Outputs.Pack(result)
+
+	case "getTotalEligibleVotesCount":
+		result := self.delayedStorage.EligibleVoteCount()
+		return method.Outputs.Pack(result)
+
+	case "getValidatorEligibleVotesCount":
+		var args ValidatorAddress
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse getValidatorEligibleVotesCount input args: ", err)
+			return nil, err
+		}
+
+		result := self.delayedStorage.GetEligibleVoteCount(&args.Validator)
+		return method.Outputs.Pack(result)
+
+	case "getValidators":
+		var args GetValidatorsArgs
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse getValidators input args: ", err)
+			return nil, err
+		}
+
+		result := self.getValidators(args)
+		return method.Outputs.Pack(result)
+
+	case "getDelegatorDelegations":
+		var args GetDelegatorDelegationsArgs
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse getDelegatorDelegations input args: ", err)
+			return nil, err
+		}
+
+		result := self.getDelegatorDelegations(args)
+		return method.Outputs.Pack(result)
+
+	case "getUndelegations":
+		var args GetDelegatorDelegationsArgs
+		if err = method.Inputs.Unpack(&args, input); err != nil {
+			fmt.Println("Unable to parse getUndelegations input args: ", err)
+			return nil, err
+		}
+
+		result := self.getUndelegations(args)
+		return method.Outputs.Pack(result)
+	}
+
+	return nil, nil
+}
+
+func (self *Contract) delegate(ctx vm.CallFrame, block types.BlockNum, args ValidatorAddress) error {
```

Comment:
I dont see any checks or errors related to the:

- Min for an address to stake: 1,000 tokens
- Min delegation for a node to have to join consensus(become validator): 1mn tokens
- Max delegation for a node to have to join consensus(become validator): 10mn tokens

Reply:
again this should be wrote down, there are no requirements like this right now. We need to have some task with all those value and add them to config

Reply:
https://docs.google.com/document/d/16xodJLO2LePq-EZVZhPw07pZwVwpD5E-sch3pAevdwg/edit

Reply:
created separate task for that